src/drf_yasg/inspectors/field.py

```diff
@@ -455,18 +455,89 @@ def decimal_return_type():
     return openapi.TYPE_STRING if rest_framework_settings.COERCE_DECIMAL_TO_STRING else openapi.TYPE_NUMBER
 
 
-raw_type_info = [
+hinting_type_info = [
     (bool, (openapi.TYPE_BOOLEAN, None)),
     (int, (openapi.TYPE_INTEGER, None)),
+    (str, (openapi.TYPE_STRING, None)),
     (float, (openapi.TYPE_NUMBER, None)),
+    (dict, (openapi.TYPE_OBJECT, None)),
     (Decimal, (decimal_return_type, openapi.FORMAT_DECIMAL)),
     (uuid.UUID, (openapi.TYPE_STRING, openapi.FORMAT_UUID)),
     (datetime.datetime, (openapi.TYPE_STRING, openapi.FORMAT_DATETIME)),
     (datetime.date, (openapi.TYPE_STRING, openapi.FORMAT_DATE)),
-    # TODO - support typing.List etc
 ]
 
-hinting_type_info = raw_type_info
+
+def get_origin_type(hint_class):
+    origin_type = hint_class.__origin__ if hasattr(hint_class, '__origin__') else None
+    return origin_type or hint_class
+
+
+def is_origin_type_subclasses(hint_class, check_class):
+    origin_type = get_origin_type(hint_class)
+    if not inspect.isclass(origin_type):
+        return False
+    return issubclass(origin_type, check_class)
+
+
+def inspect_union_hint_class(hint_class):
+    if not typing:
+        return NotHandled
+
+    if get_origin_type(hint_class) == typing.Union:
+        child_type = hint_class.__args__[0]
+        return get_basic_type_info_from_hint(child_type)
+
+    return NotHandled
```

Comment:
This is ugly, we can't just assume by default that an Union is represented by its first alternative. Since OpenAPI 2.0 doesn't provide a good way to express unions (i.e. `oneOf`), we should just ignore them - users can decide exactly how to handle them, if they so wish.

Reply:
Got it. One more reason has left to work with Union is that `Optional[T]` is represented as `Union[T, None]`. So I will rework it support the only case and leave other non-obvious untouched.

Reply:
Alright, a special case for `Optional` seems good. In that case it would probably make sense to add `x-nullable` to the type info.